### PR TITLE
Add basic turtling - needs work

### PR DESCRIPTION
**Pull request by josefwaller**

### CODE REVIEW

File: classes/coordinate.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ class Coordinate {
   toString() {
   	return 'X: ' + this.x + ', Y: ' + this.y;
   }
+  subtract(other) {
```
Comment (jwrichard) [inline classes/coordinate.js:15]:
```suggestion

  subtract(other) {
```

File: classes/coordinate.js [hunk 2]
```diff
@@ -12,6 +12,12 @@ class Coordinate {
...
+      return new Coordinate({
+          x: this.x - other.x,
+          y: this.y - other.y
+      })
```
Comment (ShannonDonovan) [inline classes/coordinate.js]:
```suggestion
      });
```

File: classes/game.js [hunk 3]
```diff
@@ -24,6 +25,44 @@ class Game{
...
+    
+    /*
+     * Return a move to turtle the snake around in a circle 
+     */
+    turtle() {
+        // Compare the coord of the head to the coord of the next segment to figure out how the snake just moved
+        const diff = this.me.head.subtract(this.me.body[1]);
+        // It should go down half its length, then turn around and go back
+        // So check that its done that
+        if (diff.x == 0) {
```
Comment (ShannonDonovan) [inline classes/game.js]: Need triple equals everywhere you have double equals

File: classes/snake.js [hunk 4]
```diff
@@ -14,6 +14,10 @@ class Snake{
             return coord.x === c.x && coord.y === c.y;
         });
     }
+    // Get the length of the snake
+    length() {
```
Comment (jwrichard) [inline classes/snake.js]: This doesn't need to be a function, could just be a property set in the constructor
`this.length = this.body.length;`

**Pull request closed without merging**